tests/node/archive.test.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const test = require('tape');
+const fsPromises = require('fs').promises;
+
+const { mArchiveFile, mgArchiveFile } = require('../../config.js');
+const { MArchive, MgArchive } = require('../../lib/archive.js');
+
+test('MArchive.put :: writes to $M_ARCHIVE_DIR/m_archive.json', async (t) => {
+  try {
+    // remove file
+    await fsPromises.unlink(mArchiveFile);
+  } catch (e) {}
+
+  await MArchive.put(['hello', 'world']);
+
+  const archive = await fsPromises.readFile(mArchiveFile);
+
+  t.deepEqual(JSON.parse(archive), ['hello', 'world']);
+  t.end();
+});
+
+test('MArchive.get :: reads from $M_ARCHIVE_DIR/m_archive.json', async (t) => {
+  await fsPromises.writeFile(mArchiveFile, JSON.stringify(['hello', 'world']));
+
+  MArchive.get((archive) => {
+    t.equal(archive, 'hello world');
+    t.end();
+  });
+});
+
+test('MgArchive.put :: writes to $M_ARCHIVE_DIR/mg_archive.json with childProcess.exec output', async (t) => {
+  try {
+    // remove file
+    await fsPromises.unlink(mgArchiveFile);
+  } catch (e) {}
```

Comment:
I see the 'remove file' comment. However, what is this try block doing?

Reply:
https://github.com/samhstn/m/blob/b8ddfa4ec3853b54965b7f7e0a98b39cd91a52c6/tests/node/archive.test.js#L33-L36

This block removes the file if it exists, otherwise don't do anything.

I added the `// remove file` comment as `fsPromises.unlink` wasn't too descriptive to me. I could instead change this to:

```node
// remove file if exists
 try { 
   await fsPromises.unlink(mgArchiveFile); 
 } catch (e) {} 
```

Reply:
Your suggestion looks good. As an addition, I suggest encapsulating the try block in a function called 'remove_file_if_exists' . I am a strong advocate of abstracting processing to clearly named functions. This approach improves readability.